modules/scripts/startup-script/variables.tf

```diff
@@ -126,6 +126,12 @@ variable "install_docker" {
   nullable    = false
 }
 
+variable "setup_raid" {
```

Comment:
Would it be useful to allow them to select the location. I kind of expect we might also want to provide additional options in the future, such as other raid levels.

I might suggest using an object:
```
settings:
  raid_local_ssds:
    local_mount: /mnt/localssd
```

Reply:
I agree that we need the ability to specify a mountpoint and that an object is a good idea. I would probably try something like:

```
variable "local_ssd_filesystem" {
  description = "Instance Image"
  type        = object({
    enable_raid = optional(bool, false)
    mount_point = optional(string, "/mnt/localssd")
  })
```

WDYT?

Reply:
Based on discussion together, I added the suggestion.
```
variable "local_ssd_filesystem" {
  description = "Format and mount filesystem from local SSD scratch disks, preserving existing data"
  type = object({
    enable     = optional(bool, false)
    fs_type    = optional(string, "ext4")
    mountpoint = optional(string, "/mnt/localssd")
  })
}
```

Reply:
For posterity, we settled upon the solution below. It addresses a couple concerns:

1. In a blueprint, a user may set `mountpoint` but forget to "enable" the usage. This solution enables the usage through any non-empty mountpoint string that begins with `/`.
2. It is open-ended with respect to alternatives to RAID (future development that we don't know yet)

```hcl
variable "local_ssd_filesystem" {
  description = "Create and mount a filesystem from local SSD disks (data will be lost if VMs are powered down without enabling migration); enable by setting mountpoint field to a valid directory path."
  type = object({
    fs_type    = optional(string, "ext4")
    mountpoint = optional(string, "")
  })

  validation {
    condition     = can(coalesce(var.local_ssd_filesystem.fs_type))
    error_message = "var.local_ssd_filesystem.fs_type must be set to a filesystem supported by the Linux distribution."
  }

  validation {
    condition     = var.local_ssd_filesystem.mountpoint == "" || startswith(var.local_ssd_filesystem.mountpoint, "/")
    error_message = "To enable local SSD filesystems, var.local_ssd_filesystem.mountpoint must be set to an absolute path to a mountpoint."
  }

  default = {
    fs_type    = "ext4"
    mountpoint = ""
  }

  nullable = false
}
```